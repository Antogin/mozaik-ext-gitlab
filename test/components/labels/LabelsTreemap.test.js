import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import { defaultTheme } from '@mozaik/ui'
import LabelsTreemap from './../../../src/components/labels/LabelsTreemap'
import { sampleProject, sampleLabels } from './sampleData'

test('should render as expected', () => {
    const tree = renderer.create(
        <ThemeProvider theme={defaultTheme}>
            <LabelsTreemap
                project={sampleProject.name}
                apiData={{ project: sampleProject, labels: sampleLabels }}
                theme={defaultTheme}
            />
        </ThemeProvider>
    )

    expect(tree).toMatchSnapshot()
})
