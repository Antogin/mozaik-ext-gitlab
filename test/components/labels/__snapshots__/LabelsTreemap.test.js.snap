// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as expected 1`] = `
<div
  className="sc-bdVaJa eydJRb"
  style={Object {}}
>
  <div
    className="sc-bwzfXH fMILzD"
  >
    <header
      className="sc-htpNat llvRri"
      style={Object {}}
    >
      <span>
        <div
          className="sc-ifAKCX bDEKRH"
        >
          <a
            href="https://gitlab.com/plouc/ploucifier"
            target="_blank"
          >
            ploucifier
          </a>
        </div>
        Labels by open issues
        <div
          className="sc-bxivhb eDwDKI"
        >
          3
        </div>
      </span>
      <div
        className="sc-EHOje dlEdwN"
      >
        <svg
          fill="currentColor"
          height="1em"
          preserveAspectRatio="xMidYMid meet"
          style={
            Object {
              "color": "#000",
              "verticalAlign": "middle",
            }
          }
          viewBox="0 0 40 40"
          width="1em"
        >
          <g>
            <path
              d="m9.5 10.5q0-1.2-0.8-1.9t-1.9-0.8-1.9 0.8-0.8 1.9 0.8 1.9 1.9 0.8 1.9-0.8 0.8-1.9z m22.7 12.2q0 1.1-0.8 1.9l-10.4 10.5q-0.8 0.8-2 0.8-1.1 0-1.9-0.8l-15.2-15.2q-0.8-0.8-1.3-2.2t-0.6-2.5v-8.8q0-1.1 0.8-1.9t1.9-0.8h8.9q1.1 0 2.5 0.5t2.1 1.4l15.2 15.2q0.8 0.8 0.8 1.9z m8.2 0q0 1.1-0.8 1.9l-10.5 10.5q-0.8 0.8-1.9 0.8-0.8 0-1.2-0.3t-1.2-1l10-10q0.8-0.8 0.8-1.9 0-1.1-0.8-1.9l-15.2-15.2q-0.8-0.8-2.1-1.4t-2.5-0.5h4.7q1.2 0 2.5 0.5t2.2 1.4l15.2 15.2q0.8 0.8 0.8 1.9z"
            />
          </g>
        </svg>
      </div>
    </header>
    <div
      className="sc-bZQynM cUinle"
      style={Object {}}
    >
      <div
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <svg
          height={400}
          width={600}
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(0,0)"
          >
            <g
              className="nivo_treemap_node"
              transform="translate(10,10)"
            >
              <rect
                fill="#F00"
                height={221}
                stroke="#F00"
                strokeWidth={0}
                width={444}
              />
              <g
                transform="translate(222,110.5) rotate(0)"
              >
                <text
                  className="nivo_treemap_node_label"
                  dy="0.5em"
                  style={
                    Object {
                      "fill": "rgb(144, 0, 0)",
                    }
                  }
                  textAnchor="middle"
                >
                  label A 21
                </text>
              </g>
            </g>
            <g
              className="nivo_treemap_node"
              transform="translate(10,233)"
            >
              <rect
                fill="#F0F"
                height={157}
                stroke="#F0F"
                strokeWidth={0}
                width={444}
              />
              <g
                transform="translate(222,78.5) rotate(0)"
              >
                <text
                  className="nivo_treemap_node_label"
                  dy="0.5em"
                  style={
                    Object {
                      "fill": "rgb(144, 0, 144)",
                    }
                  }
                  textAnchor="middle"
                >
                  label B 15
                </text>
              </g>
            </g>
            <g
              className="nivo_treemap_node"
              transform="translate(456,10)"
            >
              <rect
                fill="#00F"
                height={380}
                stroke="#00F"
                strokeWidth={0}
                width={134}
              />
              <g
                transform="translate(67,190) rotate(-90)"
              >
                <text
                  className="nivo_treemap_node_label"
                  dy="0.5em"
                  style={
                    Object {
                      "fill": "rgb(0, 0, 144)",
                    }
                  }
                  textAnchor="middle"
                >
                  label C 11
                </text>
              </g>
            </g>
          </g>
        </svg>
      </div>
    </div>
  </div>
</div>
`;
